sources/OpenMcdf.Extensions/OLEProperties/DictionaryProperty.cs

```diff
@@ -61,17 +61,31 @@ public void Read(BinaryReader br)
 
         public void Write(BinaryWriter bw)
         {
+            long curPos = bw.BaseStream.Position;
+
             bw.Write(entries.Count);
 
             foreach (KeyValuePair<uint, string> kv in entries)
             {
                 bw.Write(kv.Key);
+
                 string s = kv.Value;
                 if (!s.EndsWith("\0"))
```

Comment:
Noticed whilst testing other things that this doesn't work on .NET 6 - will update